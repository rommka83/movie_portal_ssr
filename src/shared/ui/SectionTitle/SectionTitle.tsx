import React, { HTMLAttributes } from "react";
import styles from "./sectiontitle.module.css";
import classNames from "classnames";

export const SectionTitle = React.memo<SectionTitleProps>(
	({ className, children, ...props }) => {
		return (
			<h3 {...props} className={classNames(styles.sectionTitle, className)}>
				{children}
			</h3>
		);
	}
);

type SectionTitleProps = HTMLAttributes<HTMLHeadingElement> & {
	className?: string;
};
