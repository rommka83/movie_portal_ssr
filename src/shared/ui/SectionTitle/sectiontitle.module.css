.sectionTitle {
  font-size: 24px;
  line-height: 28px;
  font-weight: 700;
  color: var(--white);
  cursor: pointer;
}
