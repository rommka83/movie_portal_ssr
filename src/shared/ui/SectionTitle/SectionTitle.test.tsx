import renderer from 'react-test-renderer';
import { SectionTitle } from './SectionTitle';

describe('проверка SectionTitle', () => {
  test('отрисовка SectionTitle', () => {
    const component = renderer.create(
      <SectionTitle>
        <span>Проверка</span>
      </SectionTitle>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
