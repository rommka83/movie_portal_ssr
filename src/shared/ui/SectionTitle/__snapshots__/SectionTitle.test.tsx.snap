// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`проверка SectionTitle отрисовка SectionTitle 1`] = `
<h3
  className="sectionTitle"
>
  <span>
    Проверка
  </span>
</h3>
`;
